rct229/rulesets/ashrae9012019/section10/section10rule14.py

```diff
@@ -0,0 +1,510 @@
+            is_zone_agg_factor_undefined_and_needed = calc_vals[
+                "is_zone_agg_factor_undefined_and_needed"
+            ]
+
+            # IF OUTCOME IS UNDETERMINED AND THESE CONDITIONS ARE TRUE, THE MOST CONSERVATIVE EFFICIENCY WAS MODELED
+            # Cases 3, 4, 7, 10
+            if len(modeled_effs_b) > 0 and all(
+                modeled_eff_b[0] == expected_eff_b[0]
+                for modeled_eff_b, expected_eff_b in zip(
+                    modeled_effs_b, expected_effs_b
+                )
+            ):
+                return MOST_CONSERVATIVE_MSG
+
+            # Case 6
+            elif (
+                hvac_system_type_b == HVAC_SYS.SYS_4
+                and modeled_high_temp_eff_b == expected_high_temp_eff_b
+                and modeled_low_temp_eff_b is None
+                and total_capacity_b is not None
+                and not is_zone_agg_factor_undefined_and_needed
+            ):
+                return UNDEFINED_LOW_TEMP_MSG
+
+            # IF OUTCOME IS UNDETERMINED AND THESE CONDITIONS ARE TRUE, THE MOST CONSERVATIVE HIGH-TEMP EFFICIENCY WAS MODELED
+            # Cases 8, 9, 11, 12
+            elif (
+                hvac_system_type_b == HVAC_SYS.SYS_4
+                and modeled_high_temp_eff_b is not None
+                and modeled_high_temp_eff_b == expected_high_temp_eff_b
+                and total_capacity_b is None
+            ):
+                return MOST_CONSERVATIVE_HP_HIGH_TEMP_MSG
+
+        def rule_check(self, context, calc_vals=None, data=None):
+            hvac_system_type_b = calc_vals["hvac_system_type_b"]
+            total_capacity_b = calc_vals["total_capacity_b"]
+            is_zone_agg_factor_undefined_and_needed = calc_vals[
+                "is_zone_agg_factor_undefined_and_needed"
+            ]
+
+            if hvac_system_type_b in SINGLE_EFF_SYS_TYPES:
+                expected_effs_b = calc_vals["expected_effs_b"]
+                modeled_effs_b = calc_vals["modeled_effs_b"]
+
+                # Case 1 and 2 both satisfied here
+                return (
+                    all(
+                        modeled_eff_b == expected_eff_b
+                        for modeled_eff_b, expected_eff_b in zip(
+                            modeled_effs_b, expected_effs_b
+                        )
+                    )
+                    and total_capacity_b is not None
+                    and not is_zone_agg_factor_undefined_and_needed
+                )
+
+            else:  # HVAC_SYS.SYS_4
+                expected_high_temp_eff_b = calc_vals["expected_high_temp_eff_b"]
+                modeled_high_temp_eff_b = calc_vals["modeled_high_temp_eff_b"]
+                expected_low_temp_eff_b = calc_vals["expected_low_temp_eff_b"]
+                modeled_low_temp_eff_b = calc_vals["modeled_low_temp_eff_b"]
+
+                # Case 5
+                return (
+                    modeled_high_temp_eff_b == expected_high_temp_eff_b
```

Comment:
It seems the `expected_high_temp_eff_b` is guaranteed not NONE value if the HVAC system is type SYS4. Could you confirm if this is correct?
I ran through the G3_5_2 functions and think that maybe the case. If so, this logic is perfect.

Reply:
Confirmed